password_security/controllers/main.py

```diff
@@ -17,7 +17,10 @@
 class PasswordSecurityHome(AuthSignupHome):
     def do_signup(self, qcontext):
         password = qcontext.get("password")
-        user = request.env.user
+        user = (
```

Comment:
the commit and the description mention the change in the copy

But this change is not mentioned

Also the title mention "...password_write_date on copy"

Could you elaborate how this search affects the purpose of this PR, please?

Reply:
I am open to separate this in two PRs. Conceptually they are different fixes but in practice they are both needed.

The case is when a user has 2FA activated. If that is the case, they get to this method as the public user, so the check in `user._check_password(password)` does not detect the use of an old password. This makes the write method run which in turn updates `password_write_date`. Only after then, a second call to `_check_password rejects` the change.

However, since `password_write_date` is not updated to today, the user is able to log in with the old (expired) password and use it for a long time after that.

BTW: I updated the PR description to match the commit message.